### PR TITLE
Select replacement

### PR DESCRIPTION
**Pull request by muhtasimhafiz**
### Description

*Please include a short description of the suggested change and the reasoning behind the approach you have chosen.*

### Link to ticket

*Please add a link to the GitHub issue being addressed by this change.*

### Type

- [ ] Fix
- [ ] Feature
- [ ] Cleanup 

### Screenshot of the result

*If your change affects the user interface, you should include a screenshot of the result with the pull request.*

### CODE REVIEW

File: app/Domain/Timesheets/Templates/partials/editTime.blade.php [hunk 1]
```diff
@@ -55,48 +55,39 @@
...
-            echo' selected="selected" ';
-        }
-        echo'>' . $row['headline'] . '</option>';
-    } ?>
-
-</select> <br />
+    <x-global::forms.select name="tickets" id="tickets" class="ticket-select" :labelText="__('label.ticket')">
+        @foreach ($tpl->get('allTickets') as $row)
+            <x-global::forms.select.select-option :class="'project_' . $row['projectId']" :data-value="$row['projectId']" :value="$row['id']" :selected="$row['id'] == $values['ticket']">
+                {!! $tpl->escape($row['headline']) !!}
```
Comment (marcelfolaron) [inline app/Domain/Timesheets/Templates/partials/editTime.blade.php:75]: Please simplify to : `{{ $row['headline'] }}`

File: app/Domain/Timesheets/Templates/showAll.blade.php [hunk 2]
```diff
@@ -132,38 +132,35 @@
...
                     <td>
-                        <x-global::forms.checkbox
-                            labelText="{{ __('label.invoiced') }}"
-                            labelPosition="right"
-                            name="invEmpl"
-                            id="invEmpl"
-                            value="on"
-                            :checked="$tpl->get('invEmpl') == '1'"
-                            onclick="submit();"
+                        <input type="checkbox" value="on" name="invEmpl" id="invEmpl" onclick="submit();"
```
Comment (marcelfolaron) [inline app/Domain/Timesheets/Templates/showAll.blade.php:135]: That needs to be reverted. Pull from the dev branch first

Comment (marcelfolaron) [reply app/Domain/Timesheets/Templates/showAll.blade.php:135]: This is true for all checkboxes. Did you do a hard overwrite?

File: app/Domain/Timesheets/Templates/showMy.blade.php [hunk 3]
```diff
@@ -205,7 +205,7 @@
             <div class="btn-group viewDropDown pull-right">
                 <x-global::actions.dropdown contentRole="secondary">
                     <x-slot:label-text>
-                        {!! __('links.week_view') !!} {!! $tpl->__('links.view') !!} <span class="caret"></span>
+                        {!! __('links.week_view') !!} {!! $tpl->__('links.view') !!}
```
Comment (marcelfolaron) [inline app/Domain/Timesheets/Templates/showMy.blade.php:208]: You can remove `$tpl->`

**Review by marcelfolaron — CHANGES_REQUESTED** [review]

File: app/Domain/Canvas/Templates/partials/canvasDialog.blade.php [hunk 4]
```diff
@@ -159,27 +174,33 @@ class="w-full"
...
-                                                echo" selected='selected' ";
-                                            }
-
-                                            echo">" . $milestoneRow->headline . "</option>"; ?>
-                                            <?php
-                                        }     ?>
-                                </select>
+                                <x-global::forms.select 
+                                    name="existingMilestone" 
+                                    class="user-select"
```
Comment (marcelfolaron) [inline app/Domain/Canvas/Templates/partials/canvasDialog.blade.php:187]: Is that class needed?

File: cache/sessions/7sEy6w3u0YVNjD3yWyxui0IHx2XnH37k00VKa8Xu
Comment (marcelfolaron) [inline cache/sessions/7sEy6w3u0YVNjD3yWyxui0IHx2XnH37k00VKa8Xu:1]: Please remove

File: cache/views/873d9700a05ae0d2e525d2cc42c1f848.php
Comment (marcelfolaron) [inline cache/views/873d9700a05ae0d2e525d2cc42c1f848.php:1]: Please remove

File: app/Views/Templates/components/forms/select/index.blade.php [hunk 5]
```diff
@@ -27,7 +27,7 @@
 
     $selectClassBuilder = 'leantime-select select '. $selectChip .' select-bordered select-'.$formHash.' '.$sizeClass.' '.$stateClass.' w-full max-w-xs input-shadow '.($leadingVisual ? 'pl-10' : '');
 
-    var_dump($selectClassBuilder);
+    // var_dump($selectClassBuilder);
```
Comment (marcelfolaron) [inline app/Views/Templates/components/forms/select/index.blade.php:30]: Can you take this out entirely?

File: app/Views/Templates/components/forms/button.blade.php [hunk 6]
```diff
@@ -29,6 +29,9 @@
         case 'link':
             $typeClass = 'btn-link text-primary hover:bg-primary/20 ';
             break;
+        case 'transparent':
```
Comment (marcelfolaron) [inline app/Views/Templates/components/forms/button.blade.php:32]: Why? Isn't this the ghost type? Or link?

File: app/Views/Templates/components/actions/dropdown/index.blade.php [hunk 7]
```diff
@@ -16,8 +14,9 @@
         'primary' => 'btn btn-primary',
         'secondary' => 'btn btn-secondary',
         'accent' => 'btn btn-accent',
-        'ghost' => 'btn btn-tertiary',
+        'ghost' => 'btn btn-tertiary text-white text-xl',
```
Comment (marcelfolaron) [inline app/Views/Templates/components/actions/dropdown/index.blade.php:17]: By hard coding the text color in here we won't be able to do dark mode. effectively. We should rarely have the need to use `text-[color]` instead it should use the emantic color names if needed

File: app/Domain/Valuecanvas/Templates/canvasDialog.blade.php [hunk 8]
```diff
@@ -63,20 +63,20 @@ class="w-full"
...
+        @if (!empty($statusLabels))
+            <x-global::forms.select name="status" id="statusCanvas" style="width: 50%" :labelText="__('label.status')">
+            </x-global::forms.select>
+            <br /><br />
+        @else
+            <input type="hidden" name="status" value="{{ $canvasItem['status'] ?? array_key_first($hiddenStatusLabels) }}" />
+        @endif
+        
+        @if (!empty($relatesLabels))
+            <x-global::forms.select name="relates" id="relatesCanvas" style="width: 50%" :labelText="__('label.relates')">
```
Comment (marcelfolaron) [inline app/Domain/Valuecanvas/Templates/canvasDialog.blade.php:75]: Please take our style tag

File: app/Domain/Users/Templates/editOwn.blade.php [hunk 9]
```diff
@@ -161,71 +161,55 @@ class="button"
...
-                                        <select name="language" id="language" style="width: 220px">
-                                            @foreach ($languageList as $languagKey => $languageValue )
-                                                <option value="{{ $languagKey }}"
-                                                        @if ($userLang == $languagKey )
-                                                            selected='selected'
-                                                         @endif >{{ $languageValue }}</option>
-                                             @endforeach
-                                        </select>
-                                    </span>
+                                    <x-global::forms.select name="language" id="language" style="width: 220px" :labelText="__('label.language')">
```
Comment (marcelfolaron) [inline app/Domain/Users/Templates/editOwn.blade.php:164]: please take out style tag

File: app/Domain/Timesheets/Templates/showMy.blade.php [hunk 10]
```diff
@@ -389,14 +406,37 @@ class="hourCell"
                                             ); ?>
                                         <?php }?>
                                     </select>
+                                    {{-- <x-global::forms.select
```
Comment (marcelfolaron) [inline app/Domain/Timesheets/Templates/showMy.blade.php:409]: If it's commented out, please remove it

File: app/Domain/Timesheets/Templates/showMy.blade.php [hunk 11]
```diff
@@ -364,6 +364,23 @@ class="hourCell"
                                             ); ?>
                                         <?php } ?>
                                     </select>
+
```
Comment (marcelfolaron) [inline app/Domain/Timesheets/Templates/showMy.blade.php:367]: Please remove commented out code

**Review by marcelfolaron — CHANGES_REQUESTED** [review]

File: app/Domain/Canvas/Templates/partials/canvasDialog.blade.php [hunk 12]
```diff
@@ -47,28 +47,43 @@ class="w-full"
...
         <input type="hidden" name="status" value="{{ $canvasItem['status'] ?? array_key_first($hiddenStatusLabels) }}" />
     @endif
 
     @if (!empty($relatesLabels))
-        <label for="relatesCanvas">{{ $tpl->__('label.relates') }}</label>
-        <select name="relates" style="width: 50%" id="relatesCanvas">
+        <x-global::forms.select 
+            name="relates" 
+            id="relatesCanvas" 
+            style="width: 50%"
```
[the remark this answers is not in the document]
Comment (marcelfolaron) [reply app/Domain/Canvas/Templates/partials/canvasDialog.blade.php:74]: `class="w-1/2"`

File: app/Domain/Tickets/Templates/includes/ticketHeader.blade.php [hunk 13]
```diff
@@ -112,28 +112,32 @@ class="wikiModal inlineEdit"
                     <!-- Static Menu Items -->
                     <x-global::actions.dropdown.item 
                         href="javascript:void(0);" 
-                        onclick="jQuery('#sprintSelect').val('all'); leantime.ticketsController.initTicketSearchUrlBuilder('{{ $currentUrlPath }}')"
+                        onclick="jQuery('#sprintSelect').val('all'); leantime.ticketsController.initTicketSearchUrlBuilder('{{ $currentUrlPath }}')" 
+                        class="text-black"
```
[the remark this answers is not in the document]
Comment (muhtasimhafiz) [reply app/Domain/Tickets/Templates/includes/ticketHeader.blade.php]: removed these classes

File: app/Domain/Timesheets/Templates/showAll.blade.php (hunk 2 above)
Comment (muhtasimhafiz) [reply app/Domain/Timesheets/Templates/showAll.blade.php]: reverted

File: app/Views/Templates/components/forms/button.blade.php (hunk 6 above)
Comment (muhtasimhafiz) [reply app/Views/Templates/components/forms/button.blade.php:39]: fixed it

**Pull request merged**